File: examples/identity/create_holder/create_holder.py
import os
import bloock
from bloock.client.identity import IdentityClient
from bloock.client.key import KeyClient
from bloock.entity.key.key_type import KeyType
from bloock.entity.key.key import Key
from bloock.entity.identity.did_method import DidMethod

# we set the API key and create a client
bloock.api_key = os.environ["API_KEY"]
# we set de identity managed API host you have deployed
bloock.identity_api_host = os.environ["IDENTITY_MANAGED_API_HOST"]

# initialize the Key Client
key_client = KeyClient()
# initialize the IdentityClient
identity_client = IdentityClient()

#create the Baby JubJub local key
local_key = key_client.new_local_key(KeyType.Bjj)

# we create the holder, passing the Baby JubJub key and holder did method
holder_key = Key(local_key)
holder_did_method = DidMethod.PolygonID

holder = identity_client.create_holder(holder_key, holder_did_method)
# will print the Holder DID public identifier. Ex: did:polygonid:polygon:main:2qCU58EJgrELSJT6EzT27Rw9DhvwamAdbMLpePztYq
print(holder.did)
# will return the DID method we chosed. Ex: identity.PolygonID
print(holder.did_method)
# will return the Key BJJ object associated to the Holder.
print(holder.key)

File: examples/identity/create_holder/create_holder.php
<?php

use Bloock\Bloock;
use Bloock\Client\KeyClient;
use Bloock\Client\IdentityClient;
use Bloock\Entity\Key\KeyType;
use Bloock\Entity\Key\Key;
use Bloock\Entity\Identity\DidMethod;

require "./vendor/autoload.php";

// we set the API key and create a client
Bloock::$apiKey = getenv("API_KEY") ?: "";
// we set de identity managed API host you have deployed
Bloock::$identityApiHost = getenv("IDENTITY_MANAGED_API_HOST") ?: "";

// initialize the Key Client
$keyClient = new KeyClient();
// initialize the IdentityClient
$identityClient = new IdentityClient();

// create the Baby JubJub local
$localKey = $keyClient->newLocalKey(KeyType::Bjj);

// we create the holder, passing the Baby JubJub key and holder did method
$holderKey = new Key($localKey);
$holderDidMethod = DidMethod::PolygonID;

$holder = $identityClient->createHolder($holderKey, $holderDidMethod);

$createdHolderDid = $holder->getDid()->getDid(); // will print the Holder DID public identifier. Ex: did:polygonid:polygon:main:2qCU58EJgrELSJT6EzT27Rw9DhvwamAdbMLpePztYq
$createdHolderDidMethod = $holder->getDid()->getDidMethod(); // will return the DID method we chosed. Ex: identity.PolygonID
$createdHolderKey = $holder->getKey(); // will return the Key BJJ object associated to the Holder.